verification/src/solidity/verifier.rs

```diff
@@ -97,3 +327,329 @@ impl Verifier {
+    const DEFAULT_DEPLOYED_BYTECODE_WITHOUT_METADATA_HASH: &'static str =  "608060405234801561001057600080fd5b50600436106100365760003560e01c8063893d20e81461003b578063a6f9dae11461005a575b600080fd5b600054604080516001600160a01b039092168252519081900360200190f35b61006d61006836600461011e565b61006f565b005b6000546001600160a01b031633146100c35760405162461bcd60e51b815260206004820152601360248201527221b0b63632b91034b9903737ba1037bbb732b960691b604482015260640160405180910390fd5b600080546040516001600160a01b03808516939216917f342827c97908e5e2f71151c08502a66d44b6f758e3ac2f1de95f02eb95f0a73591a3600080546001600160a01b0319166001600160a01b0392909216919091179055565b60006020828403121561013057600080fd5b81356001600160a01b038116811461014757600080fd5b939250505056fe";
+
+    const DEFAULT_CREATION_TX_INPUT: &'static str = concatcp!(
+        DEFAULT_BYTECODE_WITHOUT_METADATA_HASH,
+        DEFAULT_ENCODED_METADATA_HASH,
+        DEFAULT_CONSTRUCTOR_ARGS
+    );
+    const DEFAULT_DEPLOYED_BYTECODE: &'static str = concatcp!(
+        DEFAULT_DEPLOYED_BYTECODE_WITHOUT_METADATA_HASH,
+        DEFAULT_ENCODED_METADATA_HASH
+    );
+
+    #[test]
+    fn test_initialization_with_valid_data() {
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            DEFAULT_CREATION_TX_INPUT,
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(
+            verifier.is_ok(),
+            "Initialization without \"0x\" prefix failed"
+        );
+
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            &concatcp!("0x", DEFAULT_CREATION_TX_INPUT),
+            &concatcp!("0x", DEFAULT_DEPLOYED_BYTECODE),
+        );
+        assert!(verifier.is_ok(), "Initialization with \"0x\" prefix failed");
+    }
+
+    #[test]
+    fn test_initialization_with_empty_creation_tx_input_should_fail() {
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            "",
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::MetadataHashMismatch(Mismatch::expected(
+                DisplayBytes::from_str(DEFAULT_ENCODED_METADATA_HASH).unwrap()
+            ))
+        )
+    }
+
+    #[test]
+    fn test_initialization_with_creation_tx_input_as_invalid_hex_should_fail() {
```

Comment:
https://go.dev/doc/effective_go#names
> Another short example is once.Do; once.Do(setup) reads well and would not be improved by writing once.DoOrWaitUntilDone(setup). Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.

1) `test` - it's already a test, there is a `#[test]` thing on top of it and when you filter it with `cargo test` you too write `test` word. I think it is pretty clear, that this function is a test
2) `should_fail` - I think it is actually misleading. Your **test** will not fail, the code will work fine. But **one** of the functions inside will return error, yes. Usually I saw `should_fail` on the compilation tests, where compilation is actually failing, or on usual tests which are failing for some reason, and `should_fail` is a note to the testing software that this is correct. 
3) just rearrange some words to make it smaller without losing the information: 
`fn initialize_with_invalid_creation_tx_input()` 
(also I think it's pretty clear, that function with invalid input should, indeed, fail)

Reply:
Agreed with (1). Removed the prefix from test functions.
(3) Rearranged words for specified test function.

About the (2) still believe `should_fail` is a useful suffix. It may be obvious that function should fail when _invalid_ word is used in the name of the function, but it becomes unclear and requires to look inside of the test when the word _invalid_ is absent (e.g. `deserialization_of_non_cbor_hex` - is not obvious that it should fail). Adding the suffix only to some of the tests which should fail (without _invalid_) and not to add it to others (with _invalid_) is inconsistent and would confuse users. Furthermore with `test` prefix removed (as is now) it becomes clear that it is `initialization_with_creation_tx_input_as_invalid_hex` which should fail.